### PR TITLE
Fix file watcher tests

### PR DESCRIPTION
**Pull request by labkey-alan**
#### Rationale
We have a few tests failing because our file watcher code wasn't protecting against a few edge cases with null/empty lists. This PR fixes that.

#### Related Pull Requests
* n/a

#### Changes
* null check in PipelineController.CreatePipelineTriggerAction
* array empty check in PipelineJobServiceImpl.getFormSchema

### CODE REVIEW

File: pipeline/src/org/labkey/pipeline/api/PipelineJobServiceImpl.java [hunk 1]
```diff
@@ -552,6 +552,12 @@ public FormSchema getFormSchema(Container container)
             typeOptions.add(new Option<>(pipelineTriggerType.getName(), pipelineTriggerType.getName()));
         }
 
+        String typeDefaultValue = null;
+
+        if (!typeOptions.isEmpty()) {
```
Comment (labkey-klum) [inline pipeline/src/org/labkey/pipeline/api/PipelineJobServiceImpl.java]: brace styling

File: pipeline/src/org/labkey/pipeline/PipelineController.java [hunk 2]
```diff
@@ -1528,9 +1528,16 @@ public ModelAndView getView(PipelineTriggerForm form, BindException errors) thro
...
-                form.setRowId(rowId);
-                form.setReturnUrl(returnURL);
+                PipelineTriggerForm savedForm = new TableSelector(PipelineSchema.getInstance().getTableInfoTriggerConfigurations(), filter, null).getObject(PipelineTriggerForm.class);
+
+                if (savedForm != null)
+                {
+                    form = savedForm;
+                    form.setRowId(rowId);
+                    form.setReturnUrl(returnURL);
+                } else {
```
Comment (labkey-klum) [inline pipeline/src/org/labkey/pipeline/PipelineController.java]: brace styling

**Review by labkey-klum — APPROVED** [review]

File: pipeline/src/org/labkey/pipeline/api/PipelineJobServiceImpl.java (hunk 1 above)
Comment (labkey-alan) [reply pipeline/src/org/labkey/pipeline/api/PipelineJobServiceImpl.java]: JavaScript is a hard habit to break 😄
